build error with gcc-15

Types should be conflict with math. Suggest prefix with l_

https://github.com/libretro/mame2003-plus-libretro/blob/f9919eee24a17ae563d48f1804216e9f58644418/src/machine/model1_machine.c#L165

```

BUILD      libretro-mame2003_plus (target)
    TOOLCHAIN      make
Executing (target): make ARCH= CC=/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-gcc NATIVE_CC=/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-gcc LD=/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-gcc
Compiling src/machine/model1_machine.c...
src/machine/model1_machine.c:165:13: error: conflicting types for 'fadd'; have 'void(void)'
  165 | static void fadd(void)
      |             ^~~~
In file included from src/machine/model1_machine.c:8:
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/bits/mathcalls-narrow.h:24:20: note: previous declaration of 'fadd' with type 'float(double,  double)'
   24 | __MATHCALL_NARROW (__MATHCALL_NAME (add), __MATHCALL_REDIR_NAME (add), 2);
      |                    ^~~~~~~~~~~~~~~
src/machine/model1_machine.c:175:13: error: conflicting types for 'fsub'; have 'void(void)'
  175 | static void fsub(void)
      |             ^~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/bits/mathcalls-narrow.h:39:20: note: previous declaration of 'fsub' with type 'float(double,  double)'
   39 | __MATHCALL_NARROW (__MATHCALL_NAME (sub), __MATHCALL_REDIR_NAME (sub), 2);
      |                    ^~~~~~~~~~~~~~~
src/machine/model1_machine.c:186:13: error: conflicting types for 'fmul'; have 'void(void)'
  186 | static void fmul(void)
      |             ^~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/bits/mathcalls-narrow.h:33:20: note: previous declaration of 'fmul' with type 'float(double,  double)'
   33 | __MATHCALL_NARROW (__MATHCALL_NAME (mul), __MATHCALL_REDIR_NAME (mul), 2);
      |                    ^~~~~~~~~~~~~~~
src/machine/model1_machine.c:196:13: error: conflicting types for 'fdiv'; have 'void(void)'
  196 | static void fdiv(void)
      |             ^~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/bits/mathcalls-narrow.h:27:20: note: previous declaration of 'fdiv' with type 'float(double,  double)'
   27 | __MATHCALL_NARROW (__MATHCALL_NAME (div), __MATHCALL_REDIR_NAME (div), 2);
      |                    ^~~~~~~~~~~~~~~
src/machine/model1_machine.c: In function 'track_read_info':
src/machine/model1_machine.c:870:13: warning: variable 'offd' set but not used [-Wunused-but-set-variable]
  870 |         int offd;
      |             ^~~~
src/machine/model1_machine.c: In function 'track_lookup':
src/machine/model1_machine.c:1039:39: warning: declaration of 'd' shadows a previous local [-Wshadow]
 1039 |                                 float d = (a-z)*(a-z);
      |                                       ^
src/machine/model1_machine.c:1005:15: note: shadowed declaration is here
 1005 |         float d = fifoin_pop_f();
      |               ^
src/machine/model1_machine.c:1009:22: warning: variable 'behaviour' set but not used [-Wunused-but-set-variable]
 1009 |         unsigned int behaviour, entry;
      |                      ^~~~~~~~~
src/machine/model1_machine.c: In function 'groundbox_test':
src/machine/model1_machine.c:1376:18: warning: variable 'y' set but not used [-Wunused-but-set-variable]
 1376 |         float x, y, z;
      |                  ^
make: *** [Makefile:908: src/machine/model1_machine.o] Error 1

```

--- a/src/machine/model1_machine.c
+++ b/src/machine/model1_machine.c
@@ -162,7 +162,7 @@ static float ram_get_f(void)
 	return u2f(ram_data[ram_scanadr++]);
 }
 
-static void fadd(void)
+static void l_fadd(void)
 {
 	float a = fifoin_pop_f();
 	float b = fifoin_pop_f();
@@ -172,7 +172,7 @@ static void fadd(void)
 	next_fn();
 }
 
-static void fsub(void)
+static void l_fsub(void)
 {
 	float a = fifoin_pop_f();
 	float b = fifoin_pop_f();
@@ -183,7 +183,7 @@ static void fsub(void)
 	next_fn();
 }
 
-static void fmul(void)
+static void l_fmul(void)
 {
 	float a = fifoin_pop_f();
 	float b = fifoin_pop_f();
@@ -193,7 +193,7 @@ static void fmul(void)
 	next_fn();
 }
 
-static void fdiv(void)
+static void l_fdiv(void)
 {
 	float a = fifoin_pop_f();
 	float b = fifoin_pop_f();
@@ -1596,10 +1596,10 @@ struct function {
 };
 
 static struct function ftab_vf[] = {
-	{   0, fadd,            2 },
-	{   1, fsub,            2 },
-	{   2, fmul,            2 },
-	{   3, fdiv,            2 },
+	{   0, l_fadd,          2 },
+	{   1, l_fsub,          2 },
+	{   2, l_fmul,          2 },
+	{   3, l_fdiv,          2 },
 	{   5, matrix_push,     0 },
 	{   6, matrix_pop,      0 },
 	{   7, matrix_write,   12 },
@@ -1674,10 +1674,10 @@ static struct function ftab_vf[] = {
 /*   f15_swa */
 
 static struct function ftab_swa[] = {
-	{   0, fadd,            2 },
-	{   1, fsub,            2 },
-	{   2, fmul,            2 },
-	{   3, fdiv,            2 },
+	{   0, l_fadd,          2 },
+	{   1, l_fsub,          2 },
+	{   2, l_fmul,          2 },
+	{   3, l_fdiv,          2 },
 	{   5, matrix_push,     0 },
 	{   6, matrix_pop,      0 },
 	{   7, matrix_write,   12 },

